test/metabase/query_processor/pivot_test.clj

```diff
@@ -357,3 +358,49 @@
                   [nil nil 3 607]]
                  (mt/rows
                    (qp.pivot/run-pivot-query query)))))))))
+
+(deftest pivot-with-ids-or-named-fields-in-breakout
+  (mt/dataset sample-dataset
+    (testing "Query works when breakout includes field refs with IDs"
+      (let [query        (mt/mbql-query products
+                           {:breakout [$category
+                                       [:field
+                                        (mt/id :products :created_at)
+                                        {:base-type :type/DateTime :temporal-unit :month}]]
+                            :aggregation [[:count]]})
+            viz-settings (select-keys
+                          (mt/query reviews
+                                    {:pivot_table.column_split
+                                     {:rows [$rating]
+                                      :cols [$created_at]}})
+                          [:pivot_table.column_split])]
+
+        (is (schema=
+             {:status    (s/eq :completed)
+              :row_count (s/eq 156)
+              s/Keyword  s/Any}
+             (qp.pivot/run-pivot-query query
+                                       {:visualization-settings viz-settings}
+                                       (context.default/default-context))))))
+
+    (testing "If pivot_table.column_split includes includes field refs that include with field names, query still works"
+      (let [query        (mt/mbql-query products
+                           {:breakout [*category
+                                       [:field
+                                        "CREATED_AT"
+                                        {:base-type :type/DateTime :temporal-unit :month}]]
+                            :aggregation [[:count]]})
+            viz-settings (select-keys
+                          (mt/query reviews
+                                    {:pivot_table.column_split
+                                     {:rows [$rating]
+                                      :cols [$created_at]}})
+                          [:pivot_table.column_split])]
+
+        (is (schema=
+             {:status    (s/eq :completed)
+              :row_count (s/eq 156)
+              s/Keyword  s/Any}
+             (qp.pivot/run-pivot-query query
+                                       {:visualization-settings viz-settings}
+                                       (context.default/default-context))))))))
```

Comment:
This query, with named field refs in the breakout, actually still fails even if you don't call `pivot-options` because of an exception in `query_processor.util.add_alias_info`: `"Cannot determine the source table or query for Field clause [:field \"CATEGORY\" {:base-type :type/Text}]"`.

I don't quite understand why, because I think I got the field refs to look identical to the ones I have in the saved question in the repro from the issue (when created on the release branch). Pretty confused here.

Reply:
#35025 specifically mentions a nested question, you aren't creating a nested question here.